test/TestProjects/GenerateAllModels-Cadl/cadl-project.yaml

```diff
@@ -0,0 +1,3 @@
+options:
+  "@azure-tools/cadl-csharp":
+    generate-all-models: true
```

Comment:
What is the meaning of `unreferenced-types-handling: keepAll`? Also asked this at https://github.com/Azure/autorest.csharp/issues/2960#issuecomment-1556692418.